frugalos_segment/src/client/storage.rs

```diff
@@ -625,10 +625,18 @@ impl CollectFragments {
         while force || self.futures.len() + self.fragments.len() < self.data_fragments {
             force = false;
 
-            let m = track!(self
-                .spares
-                .pop()
-                .ok_or_else(|| ErrorKind::Corrupted.error(),))?;
+            let m = if let Some(m) = self.spares.pop() {
+                m
+            } else {
+                let there_are_not_enough_fragments_error =
```

Comment:
IMO, the scope of this variable is limited sufficiently and the meaning is obvious, so the redundant name is just ugly and useless.